share/python/g3mp2.py

```diff
@@ -0,0 +1,685 @@
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+import sys
+import re
+import os
+import math
+import warnings
+
+# CUimport psi4
+# CUimport p4const
+# CUimport p4util
+
+from driver import *
+
+
+# _____________________________________________________________________________
+#
+# run_g3mp2 is a Psi4 implementation of Curtiss' G3(MP2) thermochemical method
+# as described in:
+#
+#  Gaussian-3 theory using reduced Moeller-Plesset order
+#  Curtiss, Redfern, Raghavachari, Rassolov, Pople
+#  J.Chem.Phys., 110, 10(1999)
+#
+# The CCSD(T) and HLC revisions are included and described in:
+#
+#  Gaussian-3 theory using coupled cluster energies
+#  Curtiss, Raghavachari, Redfern, Baboul, Pople
+#  Chem.Phys.Lett., 314(1999) 101-107
+#
+# Daniel R. Haney 3/28/2016 --------------------------
+#
+# ------------------------------------------------------------------------
+#   TODO:
+#       Add memory check on MP(2, fc)/G3MP2Large calc
+#       to select for OUT_OF_CORE when it may exhaust memory.
+#
+
+
+ZPE_SCALE_FACTOR = 0.8929
+
+#
+# Zero Point Energy scale factors vary with applications and vibration data.
+#
+# "Computational Thermochemistry: Scale Factor Databases and Scale Factors
+# for Vibrational Frequencies Obtained from Electronic Model Chemistries,"
+#  I. M. Alecu, Jingjing Zheng, Yan Zhao, and Donald G. Truhlar*
+#  J. Chem. Theory Comput. 2010, 6, 2872.
+#  DOI: 10.1021/ct100326h
+#
+# NIST's CCCBDB, using vibrations from ~270 compounds and recalculation
+# excluding outliers, reports the scale factor as 0.899 +/- 0.025
+#
+# Curtiss' scale factor for HF/6-31G* ZPEs produced by Gaussian-94
+# is overly precise at 4 sig fig, but left unchanged.
+#
+
+
+# Convenient term:  Boltzmann kT contribution @ 298.15K in Hartrees
+#    = (k_boltzmann * T298.15 * Avogadro) / JoulePerkcal / kcalPerHartree / cal_per_kcal
+
+psi_kT = (psi_kb * 298.15 * psi_na) / psi_cal2J / psi_hartree2kcalmol / 1000.0
+
+
+g3mp2_verbose_flag = False
+use_QCISDT = False
+debug_flag = False
+
+# ________________________________________________________________________
+# ________________________________________________________________________
+
+
+def run_g3mp2(name, **kwargs):
+
+    global g3mp2_verbose_flag
+    global use_QCISDT
+    global debug_flag
+    global csv_output
+
```

Comment:
Unfortunately, globals and Psi4 do not mix well. You will have to pass slightly more data, but you can move these inside your function. Perhaps just create a dictionary called `options` that has these variables?